src/wallet/wallet.cpp

```diff
@@ -3793,7 +3791,8 @@ unsigned int CWallet::ComputeTimeSmart(const CWalletTx& wtx) const
 {
     unsigned int nTimeSmart = wtx.nTimeReceived;
     if (!wtx.hashUnset()) {
-        if (mapBlockIndex.count(wtx.hashBlock)) {
+        const CBlockIndex* pindex = LookupBlockIndex(wtx.hashBlock);
+        if (pindex) {
```

Comment:
In some of these places you could do 

```c++
if (const CBlockIndex* pindex = LookupBlockIndex(wtx.hashBlock))
```

to reduce the scope of the pindex variables.

Reply:
I didn't know you could declare inside an if statement. Cool.